api/src/utils/index.js

```diff
@@ -592,51 +592,76 @@ async function notifDepartmentChange(department, template, young, extraParams =
 }
 
 async function autoValidationSessionPhase1Young({ young, sessionPhase1, req }) {
-  // const dateDeValidation = {
-  //   2019: new Date("06/28/2019"),
-  //   2020: new Date("07/02/2021"),
-  //   2021: new Date("07/02/2021"),
-  //   "Juin 2022": new Date(2022, 5, 20, 18), //20 juin 2022 à 18h
-  //   "Juillet 2022": new Date(2022, 6, 11, 18), //11 juillet 2022 à 18h
-  // };
-  //
-  // const dateDeValidationTerminale = {
-  //   2019: new Date("06/28/2019"),
-  //   2020: new Date("07/02/2021"),
-  //   2021: new Date("07/02/2021"),
-  //   "Juin 2022": new Date(2022, 5, 22, 18), //22 juin 2022 à 18h
-  //   "Juillet 2022": new Date(2022, 6, 13, 18), //13 juillet 2022 à 18h
-  // };
   const { validationDate: dateDeValidation, validationDateForTerminaleGrade: dateDeValidationTerminale } = await getCohortValidationDate(sessionPhase1.cohort);
-
   if (!dateDeValidation || !dateDeValidationTerminale) return;
+  
+  if ( dateDeValidation === dateDeValidationTerminale ){
+    await getStatusPhase1FromValidationDate(young, dateDeValidation);
+  } else {
+    await getStatusPhase1FromValidationDateWithSpecificCase(young, dateDeValidation, dateDeValidationTerminale);
```

Comment:
Fait tes checks sur le grade avant de rentrer dans la fonction non ?
```suggestion
    await getStatusPhase1FromValidationDateWithSpecificCase(young, young.grade == "TERM" ? dateDeValidationTerminale: dateDeValidation);
```

Reply:
Et c'est un update plutot